janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java

```diff
@@ -0,0 +1,460 @@
+    public EntryList getSlice(final KeySliceQuery query, final StoreTransaction txh) throws BackendException {
+        final Future<EntryList> result = Future.fromJavaFuture(this.session.executeAsync(this.getSlice.bind()
+                .setBytes(KEY_BINDING, query.getKey().asByteBuffer())
+                .setBytes(SLICE_START_BINDING, query.getSliceStart().asByteBuffer())
+                .setBytes(SLICE_END_BINDING, query.getSliceEnd().asByteBuffer())
+                .setInt(LIMIT_BINDING, query.getLimit())
+                .setConsistencyLevel(getTransaction(txh).getReadConsistencyLevel())))
+                .map(resultSet -> fromResultSet(resultSet, this.getter));
+        result.await();
+        return result.getValue().get().getOrElseThrow(EXCEPTION_MAPPER);
+    }
+
+    @Override
+    public Map<StaticBuffer, EntryList> getSlice(final List<StaticBuffer> keys, final SliceQuery query, final StoreTransaction txh) throws BackendException {
+        final Future<Map<StaticBuffer, EntryList>> result = Future.sequence(Iterator.ofAll(keys)
+                .<Future<Tuple2<StaticBuffer, ResultSet>>> map(key -> Future.fromJavaFuture(
+                        this.session.executeAsync(this.getSlice.bind()
+                                .setBytes(KEY_BINDING, key.asByteBuffer())
+                                .setBytes(SLICE_START_BINDING, query.getSliceStart().asByteBuffer())
+                                .setBytes(SLICE_END_BINDING, query.getSliceEnd().asByteBuffer())
+                                .setInt(LIMIT_BINDING, query.getLimit())
+                                .setConsistencyLevel(getTransaction(txh).getReadConsistencyLevel())))
+                        .map(future -> Tuple.of(key, future))))
+                .map(sequence -> sequence.toJavaMap(pair -> Tuple.of(pair._1, fromResultSet(pair._2, this.getter))));
+        result.await();
+        return result.getValue().get().getOrElseThrow(EXCEPTION_MAPPER);
+    }
+
+    private static EntryList fromResultSet(final ResultSet resultSet, final GetColVal<Tuple3<StaticBuffer, StaticBuffer, Row>, StaticBuffer> getter) {
+        final Lazy<ArrayList<Row>> lazyList = Lazy.of(() -> Lists.newArrayList(resultSet));
+        // Use the Iterable overload of ofByteBuffer as it's able to allocate the byte array up front.
+        // To ensure that the Iterator instance is recreated, it is created within the closure otherwise
+        // the same iterator would be reused and would be exhausted.
+        return StaticArrayEntryList.ofStaticBuffer(() -> Iterator.ofAll(lazyList.get())
+                .<Tuple3<StaticBuffer, StaticBuffer, Row>> map(row -> Tuple.of(
+                        StaticArrayBuffer.of(row.getBytes(COLUMN_COLUMN_NAME)),
+                        StaticArrayBuffer.of(row.getBytes(VALUE_COLUMN_NAME)),
+                        row)),
+                getter);
+    }
+
+    Statement deleteColumn(final StaticBuffer key, final StaticBuffer column, final long timestamp) {
```

Comment:
These methods are used by CQLStoreManager in mutateMany. Added comments to document that.